src/campaign_server/campaign_server.cpp

```diff
@@ -102,6 +108,28 @@ time_t monotonic_clock()
 #endif
 }
 
+/* Secure password storage functions */
+bool authenticate(config& campaign, const config::attribute_value& passphrase) {
+	return util::create_hash(passphrase, campaign["passsalt"]) == campaign["passhash"];
+}
+
+std::string generate_salt(size_t len) {
+	const std::string itoa64 = "./0123456789ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz";
+	std::string salt = std::string(len, '0');
+	boost::uniform_int<> from_str(0, itoa64.length() - 1);
+	boost::variate_generator< boost::mt19937, boost::uniform_int<> > get_char(RNG, from_str);
+
+	for (size_t i = 0; i < len; i++)
```

Comment:
Control structure keywords (``for``, ``if``, etc.) are generally handled as if they were function names when it comes to spacing; additionally, the body should always be a block enclosed in braces (e.g. ``for(;;) { /* foo */ }`` instead of ``for (;;) /* foo */;``).

One line 118 above this there is also a variable that gets initialized and then never altered again. These should be declared ``const`` to denote this intention. Also, it's possible ``itoa64`` on line 117 could benefit from having ``static`` linkage?